pkg/controllers/user/nsserviceaccount/nssvcaccnt.go

```diff
@@ -0,0 +1,79 @@
+package nsserviceaccount
+
+import (
+	"context"
+	"strings"
+
+	"github.com/rancher/rancher/pkg/settings"
+	rv1 "github.com/rancher/types/apis/core/v1"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type defaultSvcAccountHandler struct {
+	serviceAccounts rv1.ServiceAccountInterface
+}
+
+func Register(ctx context.Context, cluster *config.UserContext) {
+	logrus.Debugf("Registering defaultSvcAccountHandler for checking default serviceaccount of system namespaces")
+	nsh := &defaultSvcAccountHandler{
+		serviceAccounts: cluster.Core.ServiceAccounts(""),
+	}
+	cluster.Core.ServiceAccounts("").AddHandler(ctx, "defaultSvcAccountHandler", nsh.Sync)
+}
+
+func (nsh *defaultSvcAccountHandler) Sync(key string, sa *corev1.ServiceAccount) (runtime.Object, error) {
+	if sa == nil || sa.DeletionTimestamp != nil {
+		return nil, nil
+	}
+	logrus.Debugf("defaultSvcAccountHandler: Sync serviceaccount: key=%v", key)
+
+	//handle default svcAccount of system namespaces only
+	if err := nsh.handleIfSystemNSDefaultSA(sa); err != nil {
+		logrus.Errorf("defaultSvcAccountHandler: Sync: error handling default ServiceAccount of namespace key=%v, err=%v", key, err)
+	}
+	return nil, nil
+}
+
+func (nsh *defaultSvcAccountHandler) handleIfSystemNSDefaultSA(defSvcAccnt *corev1.ServiceAccount) error {
+	//check if sa is "default"
+	if defSvcAccnt.Name != "default" {
+		return nil
+	}
+	//check if ns is a system-ns
+	namespace := defSvcAccnt.Namespace
+	if namespace == "kube-system" || namespace == "default" || !nsh.isSystemNS(namespace) {
+		return nil
+	}
+	if defSvcAccnt.AutomountServiceAccountToken != nil && *defSvcAccnt.AutomountServiceAccountToken == false {
+		return nil
+	}
+	defSvcAccntCopy := defSvcAccnt.DeepCopy()
+	automountServiceAccountToken := false
+	defSvcAccntCopy.AutomountServiceAccountToken = &automountServiceAccountToken
+	logrus.Debugf("defaultSvcAccountHandler: updating default serviceaccount key=%v", defSvcAccntCopy)
+	_, err := nsh.serviceAccounts.Update(defSvcAccntCopy)
+	if err != nil {
+		logrus.Errorf("defaultSvcAccountHandler: error updating default serviceaccnt flag for namespace: %v, err=%+v", namespace, err)
+		return err
+	}
+	return nil
+}
+
+func (nsh *defaultSvcAccountHandler) isSystemNS(namespace string) bool {
+	systemNamespacesStr := settings.SystemNamespaces.Get()
+	if systemNamespacesStr == "" {
+		return false
+	}
+
+	systemNamespaces := make(map[string]bool)
+	splitted := strings.Split(systemNamespacesStr, ",")
+	for _, s := range splitted {
+		ns := strings.TrimSpace(s)
+		systemNamespaces[ns] = true
+	}
```

Comment:
This can updated to use slice.ContainsString, no need to make a new map, split the string and pass in the values.